src/pypromice/process/value_clipping.py

```diff
@@ -27,26 +29,37 @@ def clip_values(
     cols = ["lo", "hi", "OOL"]
     assert set(cols) <= set(var_configurations.columns)
 
-    variable_limits = var_configurations[cols].dropna(how="all")
-    for var, row in variable_limits.iterrows():
+    variable_limits = var_configurations[cols].assign(
+        dependents=lambda df: df.OOL.fillna("").str.split(),
+        # Find the closure of dependents using the DependencyGraph class
+        dependents_closure=lambda df: DependencyGraph.from_child_mapping(
+            df.dependents
+        ).child_closure_mapping(),
+    )
 
+    for var, row in variable_limits.iterrows():
         if var not in list(ds.variables):
             continue
 
-        if ~np.isnan(row.lo):
-            ds[var] = ds[var].where(ds[var] >= row.lo)
-        if ~np.isnan(row.hi):
-            ds[var] = ds[var].where(ds[var] <= row.hi)
-
-        other_vars = row.OOL
-        if isinstance(other_vars, str) and ~ds[var].isnull().all():
-            for o in other_vars.split():
-                if o not in list(ds.variables):
-                    continue
-                else:
-                    if ~np.isnan(row.lo):
-                        ds[var] = ds[var].where(ds[var] >= row.lo)
-                    if ~np.isnan(row.hi):
-                        ds[var] = ds[var].where(ds[var] <= row.hi)
+        # This is a special case for rh_u_cor and rh_l_cor where values are clipped to 0 and 100.
+        if var in ["rh_u_cor", "rh_l_cor"]:
```

Comment:
I agree, this behavior is quite unexpected. I included it in the PR because it was removed last year in the same commit where I introduced the bug that caused the OOL functionality to fail. I couldn’t find any rationale behind the limits and behavior, as they were part of the initial commit where clip_variables was introduced.

I’m happy to apply the same limits

Reply:
@PennyHow Do you have an opinion on this?

Reply:
>  it was removed last year in the same commit where I introduced the bug that caused the OOL functionality to fail.
I think I removed this, probably at the same time the higher limit on RH_cor was moved from 100 to 150 (as requested by users). So this should be removed, but the modifs that caused the OOL to fail should of course be corrected.

Reply:
I have created the issue #309 to address this conversation.
The OOL functionality is currently bypassed for rh since I have removed the OOL relations